clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1317,6 +1328,8 @@ public void onFailure(RuntimeException e) {
                                     }
                                 }
                             });
+
+                            heartbeat.sentHeartbeat(now, heartbeatFuture);
```

Comment:
Related to my comment in `Heartbeat`, but I think this reordering is problematic. It is possible that the heartbeat finishes before we reach this point because it is asynchronous. If we use a boolean in `Heartbeat` instead, then we can keep the original ordering.

Reply:
Cool, that makes sense, I will change it back.